File: backend/src/main/java/com/decmoe47/todo/repository/UserRepository.java
package com.decmoe47.todo.repository;

import com.decmoe47.todo.model.entity.User;

import java.util.List;
import java.util.Optional;

public interface UserRepository extends BaseRepository<User, Long> {

    Optional<User> findByEmail(String email);

    Optional<List<User>> findByEmailContaining(String email);

    Optional<List<User>> findByNameContaining(String username);
}

File: backend/src/main/kotlin/com/decmoe47/todo/repository/UserRepository.kt
package com.decmoe47.todo.repository

import com.decmoe47.todo.model.entity.User

interface UserRepository {
    fun first(id: Long): User?

    fun firstByEmail(email: String): User?

    fun selectByEmail(email: String): List<User>

    fun selectByName(username: String): List<User>

    fun save(user: User): User

    fun update(user: User): User
}
